libraries/AP_Notify/ToshibaLED_I2C.cpp

```diff
@@ -74,20 +76,34 @@ bool ToshibaLED_I2C::hw_init()
 bool ToshibaLED_I2C::hw_set_rgb(uint8_t red, uint8_t green, uint8_t blue)
 {
     rgb = {red, green, blue};
-    _need_update = true;
+    for (uint8_t i=0; i<ARRAY_SIZE(busnum); i++) {
+        struct toshiba_dev &dev = device[i];
+        dev.need_update = true;
+    }
     return true;
 }
 
-void ToshibaLED_I2C::_timer(void)
+void ToshibaLED_I2C::timer(uint8_t i)
 {
-    if (!_need_update) {
+    struct toshiba_dev &dev = device[i];
+    if (!dev.need_update) {
         return;
     }
-    _need_update = false;
+    dev.need_update = false;
 
     /* 4-bit for each color */
     uint8_t val[4] = { TOSHIBA_LED_PWM0, (uint8_t)(rgb.b >> 4),
                        (uint8_t)(rgb.g / 16), (uint8_t)(rgb.r / 16) };
 
-    _dev->transfer(val, sizeof(val), nullptr, 0);
+    dev.dev->transfer(val, sizeof(val), nullptr, 0);
+}
+
+void ToshibaLED_I2C::timer_0(void)
+{
+    timer(0);
+}
+
+void ToshibaLED_I2C::timer_1(void)
+{
+    timer(1);
```

Comment:
This doesn't look good to me since it doesn't scale. I'd prefer to keep backend instances and the driver separate as it is in other places.  This way we can use the fact that `this` is implicitly passed on the callback and avoid the need to keep adding them like this.

Reply:
> I'd prefer to keep backend instances and the driver separate as it is in other places.

true, but that is a considerably more complex change given the current structure of AP_Notify. Are you OK with this simpler change now, or do we need to delay for a larger restructure of AP_Notify?

Reply:
another small factor is this is more efficient - otherwise we're instantiating multiple RGBLed objects. It is not a lot of extra ram and cpu, but it is definately more efficient just doing the two callbacks

Reply:
> true, but that is a considerably more complex change given the current structure of AP_Notify.

I disagree it's more complex since it's what it's done on most of the libraries.

> Are you OK with this simpler change now, or do we need to delay for a larger restructure of AP_Notify?

As long as it's done, I'm fine.  However I don't see a need to rush it in if there's a small refactor that can be done as part of the feature that improves the status quo.